src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAO.scala

```diff
@@ -475,26 +470,91 @@ class PostgresDirectoryDAO(protected val dbRef: DbReference,
 
   override def isEnabled(subject: WorkbenchSubject): IO[Boolean] = ???
 
-  override def getUserFromPetServiceAccount(petSA: ServiceAccountSubjectId): IO[Option[WorkbenchUser]] = ???
+  override def getUserFromPetServiceAccount(petSA: ServiceAccountSubjectId): IO[Option[WorkbenchUser]] = {
+    runInTransaction { implicit session =>
+      val petServiceAccountTable = PetServiceAccountTable.syntax
+      val userTable = UserTable.syntax
+
+      val loadUserQuery = samsql"""select ${userTable.resultAll}
+                from ${UserTable as userTable}
+                join ${PetServiceAccountTable as petServiceAccountTable} on ${petServiceAccountTable.userId} = ${userTable.id}
+                where ${petServiceAccountTable.googleSubjectId} = ${petSA}""" //TODO harmonize the subjectId types
```

Comment:
fix the TODO

Reply:
Resolved by changing `PetServiceAccountRecord` to have a `ServiceAccountSubjectId` instead of a `GoogleSubjectId` for the `googleSubjectId` field which is what `PetServiceAccount`s have